### PR TITLE
Shell recorder

### PR DESCRIPTION
**Pull request by markus2330**
# Purpose

Improve shell recorder.

# Checklist

Check relevant points but please do not remove entries.
For docu fixes, spell checking, and similar nothing
needs to be checked.

- [ ] commit messages are fine ("module: short statement" syntax and refer to issues)
- [ ] I added unit tests
- [ ] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)
- [ ] release notes are updated (doc/news/_preparation_next_release.md)

@markus2330 please review my pull request

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you for the extensive review!

> I assume most people would expect that an interactive shell executes a command after you pressed enter (⏎) and not after you entered EOF (Ctrl + D).

Does it for you? I tried different shells, and for them pressing enter executed the command. Ctrl+D went back to executing the script.

File: tests/shell/shell_recorder/tutorial_wrapper/README.md [hunk 1]
```diff
@@ -81,15 +108,46 @@ All check start with a comment sign (`#`).
...
+### Interactive Debugging
+
+Sometimes you want to inspect what happens at a specific line within the Shell Recorder.
+These features require to use either
+
+- `ctest --interactive-debug-mode 1`
+- run the shell recorder directly, which can be done using
+  ```
+  cd build
+  . ~e/scripts/run_dev_env
```
[the remark this answers is not in the document]
Comment (markus2330) [reply tests/shell/shell_recorder/tutorial_wrapper/README.md]: Alternative? /path/to/scripts/run_dev_env works in no shell at all :wink:

I added "(~e is the path to an Elektra checkout)" though.

**Comment by sanssecours** [comment]
> Does it for you? I tried different shells, and for them pressing enter executed the command.

I guess this is behavior is specific to `fish` then. I tried `interactive bash` (instead of `interactive`), which spawns a shell that executes commands after you press enter. The spawned version of Bash does only print my input after I press enter though, which is not very user friendly.

**Comment by markus2330** [comment]
> The spawned version of Bash does only print my input after I press enter though, which is not very user friendly.

Thank you, I updated the docu which should clarify what to do: ba8ac09670e2160cab26ed75359f68d1d518e36f

Do not hesitate to give further input if it does not work like described.

**Comment by markus2330** [comment]
jenkins build all please

**Pull request merged**